mozetl/search/daily_search_rollups.py

```diff
@@ -1,309 +1,371 @@
+    profile_share sums to 1.0.
+    """
+    shares_df = (
+        dataframe
+        .groupBy("profile")
+        .count()
+        .where("count > 1")
+        .withColumn("profile_share", F.lit(1.0) / F.col("count"))
     )
 
+    # left join and default shares to 1.0
+    return (
+        dataframe
+        .join(shares_df, "profile", "left")
+        .na.fill(1.0, ["profile_share"])
+    )
 
-TEMP_TABLE_TEMPLATE = 'rollup_temp_{}'
-
-# SELECT_ROLLUP_TEMPLATE says: gather all permutations of
-# submission_date, country, search_provider, default_provider,
-# locale, distribution_id and count:
-# 1) How many searches fall into each permutation? -> search_count
-# 2) How many profiles fall into each bucket -> profile_count
-# 3) What share of total profiles does this bucket represent? -> profile_share
-# The distinction between profile_count and profile_share is necessary
-# because on a given submission_date a single user of a profile may switch
-# default search engine (or locale or geo), or a profile may be shared by
-# multiple users on a day (causing all the other fields to vary if the
-# users are on widely distributed machines).
-#
-# Restrict to a whitelist of search_source's to avoid double counting while
-# we test our new search_count telemetry developed in:
-# https://bugzilla.mozilla.org/show_bug.cgi?id=1367554
-# https://bugzilla.mozilla.org/show_bug.cgi?id=1368089
-SELECT_ROLLUP_TEMPLATE = """
-SELECT
-  submission_date,
-  search_provider,
-  sum(search_count) as search_count,
-  country,
-  locale,
-  distribution_id,
-  default_provider,
-  count(distinct profile_day) as profile_count,
-  sum(profile_share) as profile_share
-FROM
-  {}
-WHERE
-  ((search_count > -1) OR (search_count is null))
-  AND search_source IN ('searchbar', 'urlbar', 'abouthome', 'newtab',
-                        'contextmenu', 'system', 'NO_SEARCHES')
-GROUP BY
-  submission_date, country, search_provider, default_provider, locale, distribution_id
-"""
 
-# SELECT_SHARE says: given a single profile_day with N permutations of
-# (search_provider, country, locale, distribution_id, default_provider),
-# N = an integer > 0, assign each row a profile_share of 1/N.
-# Example #1: a user switches default mid-day -> she generates two
-# rows, each with profile_count = 1 and profile_share = 0.5.
-# Example #2: a profile is cloned to ten laptops, the users of which
-# change default engines, travel across country borders, etc. ->
-# they generate N rows whose profile_counts sum to 10 and whose
-# profile_share sums to 1.0.
-SELECT_SHARE = """
-SELECT
-  {}.*,
-  share_table.profile_share
-FROM
-  {}, (
-    SELECT
-      profile_day,
-      1.0/count(*) AS profile_share
-    FROM
-      {}
-    GROUP BY profile_day)
-  share_table
-WHERE
-  {}.profile_day = share_table.profile_day
-"""
+def rollup_searches(dataframe, attributes, mode):
+    """ Gather all permutations of submission_date, country,
```

Comment:
Please update this comment to remove the explicit column list and say something like "all permutations of columns specified by the `attributes` arg"